### PR TITLE
feature - home screen prototype

### PR DESCRIPTION
**Pull request by at-haivo**
home screen prototype: 
- collection view 
- video object model
- home dummy data

### CODE REVIEW

**Comment by at-haivo** [comment]
@at-daiho please review and check it

**Comment by at-daiho** [comment]
:+1: There are no linter violations.

File: Youtube/Controllers/CategoryVC/CategoryViewController.swift [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    @IBOutlet weak var tableViewCategory: UITableView?
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        configUI()
+    }
+
+    override func configUI() {
+        tableViewCategory?.register(UINib(nibName: "TableViewCell", bundle: nil), forCellReuseIdentifier: "cell")
```
Comment (at-daiho) [inline Youtube/Controllers/CategoryVC/CategoryViewController.swift]: Please use SwiftUtils

File: Youtube/Controllers/CategoryVC/CategoryViewController.swift [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+//
+//  Created by asiantech on 6/1/17.
+//  Copyright © 2017 AsianTech. All rights reserved.
+//
+
+import UIKit
+
+class CategoryViewController: BaseViewController {
+
+    @IBOutlet weak var tableViewCategory: UITableView?
```
Comment (at-daiho) [inline Youtube/Controllers/CategoryVC/CategoryViewController.swift]: @IBOutlet private weak var tableView: UITableView!

File: Youtube/Controllers/CategoryVC/CategoryViewController.swift [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        tableViewCategory?.dataSource = self
+        tableViewCategory?.delegate = self
+    }
+}
+extension CategoryViewController:UITableViewDataSource {
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return 10
+    }
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: "cell", for: indexPath) as? TableViewCell else {
```
Comment (at-daiho) [inline Youtube/Controllers/CategoryVC/CategoryViewController.swift]: Please use SwiftUtils

File: Youtube/Controllers/HomeVC/HomeViewController.swift [hunk 4]
```diff
@@ -10,7 +10,84 @@ import UIKit
...
 class HomeViewController: BaseViewController {
 
+    @IBOutlet weak var collectionViewHome: UICollectionView?
+    fileprivate var videos: [Video] = []
+
+    // MARK: - Life cycle
     override func viewDidLoad() {
         super.viewDidLoad()
     }
+    override func configUI() {
```
Comment (at-daiho) [inline Youtube/Controllers/HomeVC/HomeViewController.swift]: new line

File: Youtube/Controllers/HomeVC/HomeViewController.swift [hunk 5]
```diff
@@ -10,7 +10,84 @@ import UIKit
...
+        self.title = "Home"
```
Comment (at-daiho) [inline Youtube/Controllers/HomeVC/HomeViewController.swift]: remove self

File: Youtube/Controllers/HomeVC/HomeViewController.swift [hunk 6]
```diff
@@ -10,7 +10,84 @@ import UIKit
...
+        self.navigationItem.leftBarButtonItem = UIBarButtonItem(image: #imageLiteral(resourceName: "ic_bt_navigation_fire"), style: .plain, target: self, action: nil)
```
Comment (at-daiho) [inline Youtube/Controllers/HomeVC/HomeViewController.swift]: remove all self. keyword if no need

File: Youtube/Controllers/HomeVC/HomeViewController.swift [hunk 7]
```diff
@@ -10,7 +10,84 @@ import UIKit
...
+        self.navigationItem.rightBarButtonItem = UIBarButtonItem(title: "Categories", style: .plain, target: self, action: nil)
+        self.navigationItem.rightBarButtonItem?.tintColor = UIColor.white
+
+        collectionViewHome?.register(UINib(nibName: "HomeCollectionViewCell", bundle: nil), forCellWithReuseIdentifier: "cell")
+        collectionViewHome?.backgroundColor = Color.black38
+        collectionViewHome?.delegate = self
+        collectionViewHome?.dataSource = self
+        videos = createVideos()
+    }
+
```
Comment (at-daiho) [inline Youtube/Controllers/HomeVC/HomeViewController.swift]: Add mark //TODO: Dummy data

File: Youtube/Controllers/HomeVC/HomeViewController.swift [hunk 8]
```diff
@@ -10,7 +10,84 @@ import UIKit
...
+    private func createVideos() -> [Video] {
+        let video1 = Video(thumbnail: "thu1", title: "thu 1", timeDescription: "1:50:20")
```
Comment (at-daiho) [inline Youtube/Controllers/HomeVC/HomeViewController.swift]: Use for loop to create data

File: Youtube/Controllers/HomeVC/HomeViewController.swift [hunk 9]
```diff
@@ -10,7 +10,84 @@ import UIKit
...
+        videos.append(video11)
+        videos.append(video12)
+        videos.append(video13)
+        videos.append(video14)
+        videos.append(video15)
+        return videos
+    }
+}
+
+extension HomeViewController:UICollectionViewDataSource {
```
Comment (at-daiho) [inline Youtube/Controllers/HomeVC/HomeViewController.swift]: add MARK

File: Youtube/Controllers/HomeVC/HomeViewController.swift [hunk 10]
```diff
@@ -10,7 +10,84 @@ import UIKit
...
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return videos.count
+    }
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
```
Comment (at-daiho) [inline Youtube/Controllers/HomeVC/HomeViewController.swift]: new line

File: Youtube/Controllers/HomeVC/HomeViewController.swift [hunk 11]
```diff
@@ -10,7 +10,84 @@ import UIKit
...
+        guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "cell", for: indexPath) as? HomeCollectionViewCell else {
```
Comment (at-daiho) [inline Youtube/Controllers/HomeVC/HomeViewController.swift]: use SwiftUtils

File: Youtube/Controllers/HomeVC/HomeViewController.swift [hunk 12]
```diff
@@ -10,7 +10,84 @@ import UIKit
...
+            return HomeCollectionViewCell()
+        }
+        cell.photoString = videos[indexPath.row].thumbnail
```
Comment (at-daiho) [inline Youtube/Controllers/HomeVC/HomeViewController.swift]: photoString?

File: Youtube/Controllers/HomeVC/cell/HomeCollectionViewCell.swift [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+//
+//  Created by asiantech on 6/1/17.
+//  Copyright © 2017 AsianTech. All rights reserved.
+//
+
+import UIKit
+
+class HomeCollectionViewCell: UICollectionViewCell {
+
+    @IBOutlet weak var imageThumbnail: UIImageView?
```
Comment (at-daiho) [inline Youtube/Controllers/HomeVC/cell/HomeCollectionViewCell.swift]: @IBOutlet private weak var thumbnailImageView: UIImageView!

File: Youtube/Controllers/HomeVC/cell/HomeCollectionViewCell.swift [hunk 14]
```diff
@@ -0,0 +1,29 @@
...
+    @IBOutlet weak var lableTime: UILabel?
+
+    var photoString: String? {
```
Comment (at-daiho) [inline Youtube/Controllers/HomeVC/cell/HomeCollectionViewCell.swift]: Private

File: Youtube/Controllers/SearchVC/SearchViewController.swift [hunk 15]
```diff
@@ -13,6 +13,19 @@ class SearchViewController: BaseViewController, UISearchBarDelegate {
...
+        createSearchBar()
+    }
+
+    private func createSearchBar() {
+        let searchBar = UISearchBar()
+        searchBar.showsCancelButton = false
+        searchBar.placeholder = "Enter your search here!"
+        searchBar.delegate = self
+        searchBar.returnKeyType = UIReturnKeyType.done
+        self.navigationItem.titleView = searchBar
```
Comment (at-daiho) [inline Youtube/Controllers/SearchVC/SearchViewController.swift]: remove self

File: Youtube/Models/Video.swift [hunk 16]
```diff
@@ -0,0 +1,25 @@
...
+//
+
+import Foundation
+class Video: NSObject {
+
+    // MARK: - property
+
+    var thumbnail: String = ""
+    var title: String = ""
+    var timeDescription: String = ""
```
Comment (at-daiho) [inline Youtube/Models/Video.swift]: timeDescription?

File: Youtube/Models/Video.swift [hunk 17]
```diff
@@ -0,0 +1,25 @@
...
+//  Created by asiantech on 6/1/17.
+//  Copyright © 2017 AsianTech. All rights reserved.
+//
+
+import Foundation
+class Video: NSObject {
+
+    // MARK: - property
+
+    var thumbnail: String = ""
```
Comment (at-daiho) [inline Youtube/Models/Video.swift]: Why thumbnail is string?

**Comment by at-haivo** [comment]
@at-daiho please review and check it

File: Youtube/Controllers/HomeVC/cell/HomeCollectionViewCell.swift [hunk 18]
```diff
@@ -10,16 +10,14 @@ import UIKit
...
-        didSet {
-            guard let photoString = photoString else {
-                return
-            }
-            self.imageThumbnail?.image = UIImage(named: photoString)
+    func config(video: Video) {
+        if let image = video.thumbnail {
+            thumbnailImageView.image = UIImage(named: image)
         }
+        lableTime?.text = video.timeDescription
```
Comment (at-daiho) [inline Youtube/Controllers/HomeVC/cell/HomeCollectionViewCell.swift]: remove ?

File: Youtube/Controllers/HomeVC/cell/HomeCollectionViewCell.swift [hunk 19]
```diff
@@ -10,16 +10,14 @@ import UIKit
 
 class HomeCollectionViewCell: UICollectionViewCell {
 
-    @IBOutlet weak var imageThumbnail: UIImageView?
-    @IBOutlet weak var lableTime: UILabel?
+    @IBOutlet private weak var thumbnailImageView: UIImageView!
+    @IBOutlet private weak var lableTime: UILabel!
```
Comment (at-daiho) [inline Youtube/Controllers/HomeVC/cell/HomeCollectionViewCell.swift]: timeLabel

File: Youtube/Controllers/HomeVC/HomeViewController.swift [hunk 20]
```diff
@@ -17,74 +17,58 @@ class HomeViewController: BaseViewController {
...
-        self.navigationItem.leftBarButtonItem?.tintColor = UIColor.white
-        self.navigationItem.rightBarButtonItem = UIBarButtonItem(title: "Categories", style: .plain, target: self, action: nil)
-        self.navigationItem.rightBarButtonItem?.tintColor = UIColor.white
+        title = "Home"
+        navigationItem.leftBarButtonItem = UIBarButtonItem(image: #imageLiteral(resourceName: "ic_bt_navigation_fire"), style: .plain, target: self, action: nil)
+        navigationItem.leftBarButtonItem?.tintColor = UIColor.white
+        navigationItem.rightBarButtonItem = UIBarButtonItem(title: "Categories", style: .plain, target: self, action: nil)
+        navigationItem.rightBarButtonItem?.tintColor = UIColor.white
 
         collectionViewHome?.register(UINib(nibName: "HomeCollectionViewCell", bundle: nil), forCellWithReuseIdentifier: "cell")
```
Comment (at-daiho) [inline Youtube/Controllers/HomeVC/HomeViewController.swift]: Please rename to collectionView and remove all ? charactor

File: Youtube/Models/Video.swift [hunk 21]
```diff
@@ -10,13 +10,11 @@ import Foundation
 class Video: NSObject {
 
     // MARK: - property
-
-    var thumbnail: String = ""
+    var thumbnail: String?
```
Comment (at-daiho) [inline Youtube/Models/Video.swift]: thumbnailPath

File: Youtube/Controllers/SearchVC/SearchViewController.swift [hunk 22]
```diff
@@ -16,7 +16,7 @@ class SearchViewController: BaseViewController, UISearchBarDelegate {
     }
 
     override func configUI() {
-        self.title = "Search"
+        title = "Search"
```
Comment (at-daiho) [inline Youtube/Controllers/SearchVC/SearchViewController.swift]: Please define

File: Youtube/Controllers/HomeVC/HomeViewController.swift [hunk 23]
```diff
@@ -17,74 +17,58 @@ class HomeViewController: BaseViewController {
...
         super.viewDidLoad()
     }
+
     override func configUI() {
-        self.title = "Home"
-        self.navigationItem.leftBarButtonItem = UIBarButtonItem(image: #imageLiteral(resourceName: "ic_bt_navigation_fire"), style: .plain, target: self, action: nil)
-        self.navigationItem.leftBarButtonItem?.tintColor = UIColor.white
-        self.navigationItem.rightBarButtonItem = UIBarButtonItem(title: "Categories", style: .plain, target: self, action: nil)
-        self.navigationItem.rightBarButtonItem?.tintColor = UIColor.white
+        title = "Home"
```
Comment (at-daiho) [inline Youtube/Controllers/HomeVC/HomeViewController.swift]: Please define

**Comment by at-daiho** [comment]
:+1: There are no linter violations.

**Pull request merged**